roles/openshift_hosted/tasks/router/router.yml

```diff
@@ -10,7 +10,7 @@
 
 - name: set_fact replicas
   set_fact:
-    replicas: "{{ openshift.hosted.router.replicas | default(router_nodes.results.results[0]['items'] | length) }}"
+    replicas: "{{ openshift.hosted.router.replicas|default(None) | get_router_replicas(router_nodes) }}"
```

Comment:
:+1: on moving to a filter plugin.

I think it might be good to switch the order of the parameters, though. It just seems odd to filter on an integer and provide a list parameter instead of filtering on a list and providing an int param to override.

```
- set_fact:
    replicas: "{{ router_nodes | get_router_replicas(openshift.hosted.router.replicas | default(None) }}"
```